FacialRecog.py

```diff
@@ -85,38 +92,74 @@ def scan_for_known_people(known_people_folder):
                 face_encodings.append(single_encoding[0])
 
                 # write to file, for performance reasons, so as to not calculate all the faces each time
-                encodedfile = np.save((os.path.join(known_people_folder, "PreEncoded"+filename) + ".npy"), single_encoding[0])
+                encodedfile = np.save((os.path.join(known_people_folder, "PreEncoded" + filename) + ".npy"),
+                                      single_encoding[0])
                 print("DEBUG: saved to document", filename)
                 print("DEBUG: saved to document", encodedfile)
 
     return names, face_encodings
 
+"""
+Gets data from KNOWN db and detects if facial encodings are the same
+"""
+def scan_for_known_people_from_db(npy_list: str):
+    all_encodings = []
+    data = get_all_known("not_api_call")
+
+    for i in data:
+        # decode b64 np array and save that into a list
+        foo = i['npy'].strip('][').split(', ')
+        for j in range(len(foo)):
+            foo[j] = float(foo[j])
+        all_encodings.append(foo)
+
+    all_encodings = np.array(all_encodings)
+    npy_array = np.array(npy_list)
+    detected_faces = face_recognition.compare_faces(all_encodings, npy_array)
+
+    return detected_faces
+
 
 '''
 Helper function for image pre-processing
 '''
+
+
 def image_files_in_folder(folder):
     # following code snippet from face_recognition
     return [os.path.join(folder, f) for f in os.listdir(folder) if re.match(r'.*\.(jpg|jpeg|png)', f, flags=re.I)]
 
+
 '''
 Function to add the file path of an image to database
+Currently deprecated
 '''
-def add_facial_data(name: str, image_path: str):
-    data = {"name": name, "image": image_path}
-    return add_users(data)
+# def add_facial_data(name: str, image_path: str):
+#     data = {"name": name, "image": image_path}
+#     return add_users(data)
 
 '''
 Updates the known persons in the frame from the facial recog script
 '''
-def add_to_known_stream(name: str, encoded_image):
-    data = {"name": name, "img": encoded_image}
+
+
+def add_to_known_stream(name: str, encoded_image: str, encoded_encoding: str):
+    data = {"name": name, "img": encoded_image, "npy": encoded_encoding}
     return add_known_to_stream(data)
 
+
 '''
 Updates the unknown persons in the frame from facial recog script
 '''
+
+
 def add_to_unknown_stream(encoded_image: str):
     data = {"img": encoded_image}
     return add_unknown_to_stream(data)
 
+
+def decode_image(encoded_image: str):
```

Comment:
when is this used?

Reply:
when did this get added/what is the point? cough..cough...comments...cough...cough